src/__tests__/Login.test.js

```diff
@@ -0,0 +1,60 @@
+import React from "react";
+import { renderWithReduxAndRouter as render } from "../utils/renderWithReduxAndRouter.js";
+import { BrowserRouter as Router } from "react-router-dom";
+import {
+  fireEvent,
+  wait,
+  getAllByText,
+  getAllByAltText,
+} from "@testing-library/react";
+import Login from "../components/Login";
+import axios from "axios";
+import { Provider } from "react-redux";
+
```

Comment:
Testing isn't the easiet thing to begin with. 
Adding in ReactRouter And a Redux Store make it that much more complicated.

But we used them and had to test with them to be accurate in how our app would be encountered by an enduser.

Great job  sorting this out and getting it running with the Provider and the BrowserRouter.

Kudos! 🦘